crates/kitsune_p2p/types/src/agent_info.rs

```diff
@@ -0,0 +1,293 @@
+//! Data structures to be stored in the agent/peer database.
+
+use crate::bin_types::*;
+use crate::dht_arc::DhtArc;
+use crate::*;
+
+use crate::tx2::tx2_utils::TxUrl;
+
+/// A list of Urls.
+pub type UrlList = Vec<TxUrl>;
+
+/// agent_info helper types
+pub mod agent_info_helper {
+    use super::*;
+
+    #[allow(missing_docs)]
+    #[derive(Debug, serde::Serialize, serde::Deserialize)]
+    pub struct AgentMetaInfoEncode {
+        pub dht_storage_arc_half_length: u32,
+    }
+
+    #[allow(missing_docs)]
+    #[derive(Debug, serde::Serialize, serde::Deserialize)]
+    pub struct AgentInfoEncode {
+        pub space: Arc<KitsuneSpace>,
+        pub agent: Arc<KitsuneAgent>,
+        pub urls: UrlList,
+        pub signed_at_ms: u64,
+
+        /// WARNING-this is a weird offset from the signed_at_ms time!!!!
+        pub expires_after_ms: u64,
+
+        #[serde(with = "serde_bytes")]
+        pub meta_info: Box<[u8]>,
+    }
+
+    #[allow(missing_docs)]
+    #[derive(Debug, serde::Deserialize)]
+    pub struct AgentInfoSignedEncode {
+        pub agent: Arc<KitsuneAgent>,
+        pub signature: Arc<KitsuneSignature>,
+        #[serde(with = "serde_bytes")]
+        pub agent_info: Box<[u8]>,
+    }
+
+    #[allow(missing_docs)]
+    #[derive(Debug, serde::Serialize)]
+    pub struct AgentInfoSignedEncodeRef<'lt> {
+        pub agent: &'lt Arc<KitsuneAgent>,
+        pub signature: &'lt Arc<KitsuneSignature>,
+        #[serde(with = "serde_bytes")]
+        pub agent_info: &'lt [u8],
+    }
+}
+
+use agent_info_helper::*;
+
+/// The inner constructable AgentInfo struct
+pub struct AgentInfoInner {
+    /// The space this agent info is relevant to.
+    pub space: Arc<KitsuneSpace>,
+
+    /// The pub key of the agent id this info is relevant to.
+    pub agent: Arc<KitsuneAgent>,
+
+    /// The storage arc currently being published by this agent.
+    pub storage_arc: DhtArc,
+
+    /// List of urls the agent can be reached at, in the agent's own preference order.
+    pub url_list: UrlList,
+
+    /// The absolute unix ms timestamp that the agent info was signed at,
+    /// according to the agent's own clock.
+    pub signed_at_ms: u64,
+
+    /// The absolute unix ms timestamp this info will expire at,
+    /// according to the agent's own clock.
+    /// Note--the encoded bootstrap version of this struct uses a weird
+    /// offset from the signed time... but this value here is the more
+    /// intuitive absolute value.
+    pub expires_at_ms: u64,
+
+    /// Raw bytes of agent info signature as kitsune signature.
+    pub signature: Arc<KitsuneSignature>,
+
+    /// the raw encoded bytes sent to bootstrap server or use for sig verify.
+    pub encoded_bytes: Box<[u8]>,
```

Comment:
Just curious why a `Box<[u8]>` and not a `Vec<u8>`?

Reply:
Mostly as a hint that this is static, shouldn't ever grow / shrink.